### ISSUE TITLE
Question: No fix cluster

### ISSUE THREAD

**Issue by tdeheurles** [coreos/etcd#3452]
**I'm trying to have that architecture :**

- Start a CoreOS
   - if first : bootstrap a single node etcd cluster
   - else : join the cluster

- when a node deconnect
   - after a few moment the cluster remove it

I know that the good way should be to have a 3 static nodes cluster and proxy to this but the goal is to start VitrualMachine that will join together automatically. I'd like to not impose a fix cluster.
My feeling is that on a local network, partitions are not the main problem. So when a node go done, it certainly have just been stopped. 
I understand that it's very fragile as it can have problem with cable disconnection or lots of other things that I don't expect. `=> But for now I'd like to just test how it will evolve.`

**So as expected, I'm confronted to the raft problem when stopping machines.**
- expect no problem on 3 nodes cluster and more : 
   - Following fault tolerance table, I will be able to remove the node when a member stop as the cluster will remain healthy (as long as all the virtual machines are not stopped at the same time).
- blocked on 2 nodes cluster :
   - If I stop a machine, the cluster becomes unhealthy (as expected by raft).
   - I cannot remove the stopped node. (protection for partition etc).

My feeling here is :
- I can try to do simple test like : `curl www.google.com is 200` so I should not be partitioned.
- I'm in experimentation and failure is totally ok

So :
- Is there a way to force remove a node ?
- If not, can I reboot a new cluster without losing some KV ? In my case I'd like to maintain /registry for example to maintain a kubernetes. Or should I do it manually ?

I totally understand that going under 3 nodes is playing with fire and any failure will stop everything.

**Comment by xiang90** [comment]
> Is there a way to force remove a node ?

No. 

> If not, can I reboot a new cluster without losing some KV ? In my case I'd like to maintain /registry for example to maintain a kubernetes. Or should I do it manually ?

You can read through https://github.com/coreos/etcd/blob/master/Documentation/runtime-reconf-design.md#permanent-loss-of-quorum-requires-new-cluster. I believe it answers all of your questions.

**Comment by tdeheurles** [comment]
Perfect, thank you @xiang90

**Comment by xiang90** [comment]
@tdeheurles Not a problem at all. Let us know if we can improve our docs!

**Comment by tdeheurles** [comment]
I think the documentation is very clear on that point.
I was using consul at the same time for the same reason. Consul allow to remove nodes from 2 nodes to 1 node. So I tried to do the same with etcd.

But after hours of removing/adding nodes, the consul cluster breaks, `sometimes`.

- You are clear, `we do not go in that way`. 
- Consul try but seems to fail `sometimes`.

I have tried to clean the etcd cluster member to reboot the cluster. It's not concluent for now as I should not be doing the good operations. I stop that for the moment and will come back later, maybe with proposition on the part of the documentation.
